### PR TITLE
Refactored fixtures regeneration

### PR DESCRIPTION
**Pull request by ssbarnea**

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -23,6 +23,8 @@ deps =
 commands =
   # safety measure to assure we do not accidentally run tests with broken dependencies
   {envpython} -m pip check
+  # Ensure that our fixtures are kept updated
+  {envpython} tools/regenerate-fixtures.py
```
Comment (cognifloyd) [inline tox.ini:27]: Doing this will mean that people that use `tox` would need to have `prettier` on their PATH (vs only installed by pre-commit)

Comment (ssbarnea) [reply tox.ini:28]: Yep. Precommit does not allow us to create a hook that uses two languages (node and python) so we need to take care of installing prettier ourselves.

File: tools/regenerate-fixtures.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+from pathlib import Path
+
+from ansiblelint.yaml_utils import FormattedYAML
+
+"""Re-generate formatting fixtures with prettier and internal formatter.
+
+Pass ``--regenerate-formatting-fixtures`` to run this and skip all other tests.
+This is a "test" because once fixtures are regenerated,
+we run prettier again to make sure it does not change files formatted
+with our internal formatting code.
```
Comment (cognifloyd) [inline tools/regenerate-fixtures.py:13]:
```suggestion
Once fixtures are regenerated, we run ``prettier`` again to make sure that
``prettier`` does not change files formatted with our internal formatting code.
```

File: tox.ini (hunk 1 above)
Comment (cognifloyd) [reply tox.ini:29]: I would rather not force people to install npm + prettier for most local development, but let it run in CI to catch issues when people don't have it installed locally.

Comment (cognifloyd) [reply tox.ini:29]: (pre-commit hides the fact that it's getting installed)

**Pull request closed without merging**